src/components/bulk-email-tool/bulk-email-form/BulkEmailForm.jsx

```diff
@@ -85,62 +135,95 @@ function BulkEmailForm(props) {
   };
 
   const createEmailTask = async () => {
-    const emailData = new FormData();
     if (validateEmailForm()) {
       setEmailFormStatus(() => FORM_SUBMIT_STATES.PENDING);
-      emailData.append('action', 'send');
-      emailData.append('send_to', JSON.stringify(selectedRecipients));
-      emailData.append('subject', editor.emailSubject);
-      emailData.append('message', editor.emailBody);
-      if (isScheduled) {
-        emailData.append('schedule', new Date(`${editor.scheduleDate} ${editor.scheduleTime}`).toISOString());
-      }
-      let data;
-      try {
-        data = await postBulkEmail(emailData, courseId);
-      } catch (e) {
-        setEmailFormStatus(FORM_SUBMIT_STATES.ERROR);
-        return;
+      let response;
+      if (editor.editMode) {
+        const editedEmail = formatDataForFormAction(FORM_ACTIONS.PATCH);
+        response = await dispatch(editScheduledEmailThunk(editedEmail, courseId, editor.schedulingId));
+      } else {
+        const emailData = formatDataForFormAction(FORM_ACTIONS.POST);
+        response = await dispatch(postBulkEmailThunk(emailData, courseId));
       }
-      if (data.status === 200) {
+      if (response?.status === 200) {
         if (isScheduled) {
           setEmailFormStatus(FORM_SUBMIT_STATES.COMPLETE_SCHEDULE);
         } else {
           setEmailFormStatus(FORM_SUBMIT_STATES.COMPLETE);
         }
-        resetEmailForm();
+      } else {
+        setEmailFormStatus(FORM_SUBMIT_STATES.ERROR);
       }
+      dispatch(getScheduledBulkEmailThunk(courseId, 1));
+      delayedEmailFormReset();
     }
   };
 
   useEffect(() => {
+    if (!!editor.scheduleDate || !!editor.scheduleTime) {
+      toggleScheduled(true);
+    }
     if (isScheduled) {
-      setEmailFormStatus(FORM_SUBMIT_STATES.SCHEDULE);
+      if (editor.editMode) {
+        setEmailFormStatus(FORM_SUBMIT_STATES.RESCHEDULE);
```

Comment:
Once the email form has entered `editMode`, does the user have a way to cancel or back out?

Reply:
unchecking the scheduling box, but other than that, no. I could add a cancel button? maybe next to the send button?

Reply:
I think a cancel button may be more intuitive to the users of the tool. Think this is something we need to run past Gabe?

I'd hate to delay this any further, and the PR already has a lot of changes. Maybe it can be a fast followup?